src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1208 @@
+            return feats
+
+        if self.downstream_fusion_method in ["elmo", "elmo_tensor"]:
+            layer_weights = F.softmax(self.learned_layer_weights, dim=0)
+            if self.downstream_fusion_method == "elmo":
+                feats = torch.sum(layer_weights.view(-1, 1, 1) * feats, dim=0)
+            else:
+                feats = torch.sum(layer_weights.view(-1, 1, self.hidden_size * 2) * feats, dim=0)
+        elif self.downstream_fusion_method in ["gated", "gated_meter"]:
+            if self.downstream_fusion_method == "gated":
+                layer_gate = F.softmax(self.gate(feats), dim=0)
+                feats = torch.sum(layer_gate * feats, dim=0)
+            else:
+                layer_gate = F.softmax(self.gate(feats[:-1]), dim=0)
+                feats = feats[-1] + torch.sum(layer_gate * feats[:-1], dim=0)
+        elif self.downstream_fusion_method == "max_pooling":
+            feats = torch.max(feats, dim=0)[0]
+        elif self.downstream_fusion_method == "mean_pooling":
+            feats = torch.mean(feats, dim=0)
+        elif self.downstream_fusion_method == "max_mean_pooling":
+            max_feats = torch.max(feats, dim=0)[0]
+            mean_feats = torch.mean(feats, dim=0)
+            feats = torch.cat([max_feats, mean_feats], dim=-1)
+        elif self.downstream_fusion_method in ["lstm_uni", "lstm_bi"]:
+            feats = self.lstmpooler_dropout(self.lstmpooler(feats)[0][-1])
+
+        return self.LayerNorm(feats)
+
+
+class BridgeTowerPooler(nn.Module):
+    def __init__(self, hidden_size):
+        super().__init__()
+        self.dense = nn.Linear(hidden_size, hidden_size)
+        self.activation = nn.Tanh()
+
+    def forward(self, hidden_states):
+        first_token_tensor = hidden_states[:, 0]
+        pooled_output = self.dense(first_token_tensor)
+        pooled_output = self.activation(pooled_output)
+        return pooled_output
+
+
+class BridgeTowerBertLinkLayer(nn.Module):
+    def __init__(self, config, link_tower_type):
+        super().__init__()
+        self.chunk_size_feed_forward = config.chunk_size_feed_forward
+        self.seq_len_dim = 1
+        self.link_tower_type = link_tower_type
+        # Like ViL-BERT and LXMERT, we don't use the self-attention and just use the cross-attention instead (optional with ffn).
+        self.crossattention = BertAttention(config)
+        if self.link_tower_type == "cross_attention_ffn":
+            self.intermediate = BertIntermediate(config)
+            self.output = BertOutput(config)
+
+    def forward(
+        self,
+        hidden_states,
+        encoder_hidden_states,
+        attention_mask=None,
+        encoder_attention_mask=None,
+        output_attentions=False,
+    ):
+        cross_attention_outputs = self.crossattention(
+            hidden_states,
+            attention_mask,
+            None,
+            encoder_hidden_states,
+            encoder_attention_mask,
+            None,
```

Comment:
why these arguments are hardcoded to `None`? 
If it is similar as Bert, you can add `head_mask=None`

Reply:
This has been fixed.